circuits/cpp/src/aztec3/utils/array.hpp

```diff
@@ -98,6 +115,33 @@ template <size_t SIZE> NT::boolean is_array_empty(std::array<NT::fr, SIZE> const
     return true;
 };
 
+/**
+ * Inserts the `source` array at the first zero-valued index of the `target` array.
+ * Ensures that all values after the first zero-valued index are zeros too.
+ * Fails if the `source` array is too large vs the remaining capacity of the `target` array.
+ */
+template <size_t size_1, size_t size_2, typename T>
+void push_array_to_array(std::array<T, size_1> const& source, std::array<T, size_2>& target)
+{
```

Comment:
Feels like we should be able to merge this with its `fr` workalike :thinking: Could we just do something like `if T == NT::fr check == NT::fr(0) else check is_empty()`?